internal/cluster/statemachine.go

```diff
@@ -14,6 +15,21 @@ const (
 	TransitionTypeRefreshStatus          = "RefreshStatus"
 )
 
+func SequentialPostTransitions(transitions ...stateswitch.PostTransition) stateswitch.PostTransition {
+	return func(stateSwitch stateswitch.StateSwitch, args stateswitch.TransitionArgs) error {
+		var ret error
+
+		for _, transition := range transitions {
```

Comment:
I think we should use the term `postTransition` and `postTransitions` here
as we should make it clear to the reader that these are not actually transitions.